geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/ExportDataFunction.java

```diff
@@ -59,17 +60,22 @@ public void execute(FunctionContext context) {
         } else {
           snapshotService.save(exportFile, SnapshotFormat.GEMFIRE);
         }
+
         String successMessage = CliStrings.format(CliStrings.EXPORT_DATA__SUCCESS__MESSAGE,
             regionName, exportFile.getCanonicalPath(), hostName);
-        context.getResultSender().lastResult(successMessage);
+        result = new CliFunctionResult(context.getMemberName(), CliFunctionResult.StatusState.OK,
+            successMessage);
       } else {
-        throw new IllegalArgumentException(
+        result = new CliFunctionResult(context.getMemberName(), CliFunctionResult.StatusState.ERROR,
             CliStrings.format(CliStrings.REGION_NOT_FOUND, regionName));
       }
-
     } catch (Exception e) {
       context.getResultSender().sendException(e);
+      result = new CliFunctionResult(context.getMemberName(), CliFunctionResult.StatusState.ERROR,
```

Comment:
do we still need this catch block? I thought if we are returning CliFunctionResult, we don't anymore. Oh, it's better to extend CliFunction.